packages/api/cms-api/src/user-permissions/user.resolver.ts

```diff
@@ -25,18 +24,4 @@ export class UserResolver {
         const [users, totalCount] = await this.userService.findUsers(args);
         return new PaginatedUserList(users, totalCount, args);
     }
-
-    @Mutation(() => Boolean)
-    @RequiredPermission("impersonation", { skipScopeCheck: true })
-    async userPermissionsStartImpersonation(@Args("userId", { type: () => String }) userId: string, @Context() request: Request): Promise<boolean> {
-        this.userService.setImpersonatedUser(userId, request);
```

Comment:
Is it possible that this error `Cannot remove headers after they are sent to the client` only occurs because we don't await the async `setImpersonatedUser` that manipulates the response?

(Though this should also happen locally, I guess)